### PR TITLE
Fixes #31191 - use correct status when report scheduling fails

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: app/controllers/api/v2/report_templates_controller.rb [hunk 1]
```diff
@@ -162,7 +162,7 @@ def schedule_report
           process_resource_error
         end
       rescue => e
-        render_error 'standard_error', :status => :internal_error, :locals => { :exception => e }
+        render_error 'standard_error', :status => :internal_server_error, :locals => { :exception => e }
```
Comment (evgeni) [inline app/controllers/api/v2/report_templates_controller.rb]: Or should this rather be something like

```ruby
render_error :custom_error, :status => :unprocessable_entity,
                     :locals => { :message => _("Scheduling Report template failed for: %s." % e.message) }
```

Comment (tbrisker) [reply app/controllers/api/v2/report_templates_controller.rb]: makes sense, that would be on par with the generate action

**Review by tbrisker — APPROVED** [review]
Thanks @evgeni !

**Comment by tbrisker** [comment]
2.2 - 024116624
2.1 - ea27aadfd

**Pull request merged**